homeassistant/bootstrap.py

```diff
@@ -113,15 +113,17 @@ def async_from_config_dict(config: Dict[str, Any],
         yield from hass.async_add_job(loader.prepare, hass)
 
     # Make a copy because we are mutating it.
-    new_config = OrderedDict()
-    for key, value in config.items():
-        new_config[key] = value or {}
-    config = new_config
+    config = OrderedDict(config)
 
     # Merge packages
     conf_util.merge_packages_config(
         config, core_config.get(conf_util.CONF_PACKAGES, {}))
 
+    # Ensure we have no None values after merge
+    for key, value in config.items():
+        if not value:
```

Comment:
[PEP8 recommends](https://www.python.org/dev/peps/pep-0008/#id51):

> Also, beware of writing `if x` when you really mean `if x is not None` -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

```python
for key, value in config.items():
    if value is None:
        config[key] = {}
```

Reply:
I think my version might be equal to: `config[key] = value or {}`, which is the correct behaviour

Reply:
Or let correct by saying current behaviour

Reply:
Ok, but then the comment should rather be: 

```python
# Ensure we have no None, empty or false values after merge
for key, value in config.items():
    if not value:
        config[key] = {}
```